realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -69,20 +68,33 @@
 
     final long threadId;
     protected final RealmConfiguration configuration;
+    // Which RealmCache is this Realm associated to. It could be null if the Realm instance is opened without putting it
```

Comment:
It is null if the Realm instance is opened without being put into a cache. It is also null if the Realm is closed.